### PR TITLE
Feature: Add Person, Student and Teacher classes 

### PR DESCRIPTION
**Pull request by luftedar**
### Hello :wave: 
# Feature: Add Person, Student and Teacher classes 

In this update: 

- Created Person class.
- Created Student class which inherits from Person.
- Created Teacher class which inherits from Person.

Have a nice review :+1:

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (mikethreels) [inline student.rb]: - [x] in the initialize method of this class `age`, `name` and `parent_permission` should be passed as arguments as well,  please remember that `name` and `parent_permission` should both have a default value as well.

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super(age, name, parent_permission)
+    @specialization = specialization
+  end
```
Comment (mikethreels) [inline teacher.rb]: - [x] in the initialize method of this class `age`, `name` and `parent_permission` should be passed as arguments as well,  please remember that `name` and `parent_permission` should both have a default value as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Orcun, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._
- [ ] because your readme file wasn't changed in this PR I will leave my suggestion here instead of in the inline comments. just copying and pasting the template from Microverse is not enough to make your project look personal and professional, please change all the placeholder information to information related to you and your project and remove the parts which are not applicable for your project right now.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Orcun, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
- [ ] because your readme file wasn't changed in this PR I will leave my suggestion here instead of in the inline comments. just copying and pasting the template from Microverse is not enough to make your project look personal and professional, please change all the placeholder information to information related to you and your project and remove the parts which are not applicable for your project right now.

### It seems like you missed my comment regarding the readme file in the previous review, in case you have any questions regarding this comment please reach out to me via Slack. you'll be able to find me under my full name "Michael Threels"

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @luftedar, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**